google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeployToolsMenuAction.java

```diff
@@ -69,4 +78,33 @@ public void actionPerformed(AnActionEvent event) {
       }
     }
   }
+
+  /**
+   * Determines if the project has at least one module with an App Engine standard or flexible
+   * facet. If it does not, then a notification balloon is shown.
+   */
+  @VisibleForTesting
+  boolean isAppEngineProjectCheck(@NotNull Project project) {
+    AppEngineProjectService projectService = AppEngineProjectService.getInstance();
+    boolean hasAppEngineFacet =
+        Stream.of(ModuleManager.getInstance(project).getModules())
+            .anyMatch(module -> projectService.hasAppEngineStandardFacet(module)
+                || projectService.hasAppEngineFlexFacet(module));
+
+    if (!hasAppEngineFacet) {
```

Comment:
What do you think about putting the notification into code into the actionPerformed method?  I tend to prefer confining side effects of an action to the method 'driving' it, instead of scattering it across the call chain.  

Another option is to just rename the  method to make it really clear what effect it has. e.g. `notifyErrorIfNotAppEngineProject`, but then it's not immediately obvious what the boolean return value is supposed to be.